direct/tpm2/commit_test.go

```diff
@@ -0,0 +1,160 @@
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+
+	defer thetpm.Close()
+
+	password := []byte("hello")
+
+	primary := CreateLoaded{
+		ParentHandle: tpm.RHEndorsement,
+		InSensitive: tpm2b.SensitiveCreate{
+			Sensitive: tpms.SensitiveCreate{
+				UserAuth: tpm2b.Auth{
+					Buffer: password,
+				},
+			},
+		},
+		InPublic: tpm2b.Template{
+			Template: tpmt.Public{
+				Type:    tpm.AlgECC,
+				NameAlg: tpm.AlgSHA1,
+				ObjectAttributes: tpma.Object{
+					FixedTPM:            true,
+					FixedParent:         true,
+					SensitiveDataOrigin: true,
+					UserWithAuth:        true,
+					Decrypt:             true,
+					Restricted:          true,
+				},
+				Parameters: tpmu.PublicParms{
+					ECCDetail: &tpms.ECCParms{
+						Symmetric: tpmt.SymDefObject{
+							Algorithm: tpm.AlgAES,
+							KeyBits: tpmu.SymKeyBits{
+								AES: helpers.NewKeyBits(128),
+							},
+							Mode: tpmu.SymMode{
+								AES: helpers.NewAlgID(tpm.AlgCFB),
+							},
+						},
+						CurveID: tpm.ECCNistP256,
+						KDF: tpmt.KDFScheme{
+							Scheme: tpm.AlgNull,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	rspCP, err := primary.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("could not create key: %v", err)
+	}
+
+	create := CreateLoaded{
+		ParentHandle: AuthHandle{
+			Handle: rspCP.ObjectHandle,
+			Name:   rspCP.Name,
+			Auth:   PasswordAuth(password),
+		},
+		InSensitive: tpm2b.SensitiveCreate{
+			Sensitive: tpms.SensitiveCreate{
+				UserAuth: tpm2b.Auth{
+					Buffer: password,
+				},
+			},
+		},
+		InPublic: tpm2b.Template{
+			Template: tpmt.Public{
+				Type:    tpm.AlgECC,
+				NameAlg: tpm.AlgSHA1,
+				ObjectAttributes: tpma.Object{
+					FixedTPM:            true,
+					FixedParent:         true,
+					UserWithAuth:        true,
+					SensitiveDataOrigin: true,
+					SignEncrypt:         true,
+				},
+				Parameters: tpmu.PublicParms{
+					ECCDetail: &tpms.ECCParms{
+						Symmetric: tpmt.SymDefObject{
+							Algorithm: tpm.AlgNull,
+						},
+						Scheme: tpmt.ECCScheme{
+							Scheme: tpm.AlgECDAA,
+							Details: tpmu.AsymScheme{
+								ECDAA: &tpms.SigSchemeECDAA{
+									HashAlg: tpm.AlgSHA1,
+									Count:   1,
+								},
+							},
+						},
+						CurveID: tpm.ECCBNP256,
+						KDF: tpmt.KDFScheme{
+							Scheme: tpm.AlgNull,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	rspC, err := create.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("could not create key: %v", err)
+	}
+
+	commit := Commit{
+		SignHandle: AuthHandle{
+			Handle: rspC.ObjectHandle,
+			Name:   rspC.Name,
+			Auth:   PasswordAuth(password),
+		},
+		P1: tpm2b.ECCPoint{
+			Point: tpms.ECCPoint{
+				X: tpm2b.ECCParameter{
+					Buffer: []byte{1},
+				},
+				Y: tpm2b.ECCParameter{
+					Buffer: []byte{2},
+				},
+			},
+		},
+		S2: tpm2b.SensitiveData{
+			Buffer: []byte{},
+		},
+		Y2: tpm2b.ECCParameter{
+			Buffer: []byte{},
+		},
+	}
+
+	_, err = commit.Execute(thetpm)
```

Comment:
Testing the output response with some off-tpm validation would be good here.

Reply:
Umm...Any good idea to test this output...?

I have just an idea that implements full ECDAA Setup/Join for this, but it's complex and we should avoid it.

Reply:
I think @chrisfenner is a better person to discuss this with, however he is out at the moment and will be back next week. In the meantime, I will add to be a reviewer.

Reply:
I'm new to ECDAA signing, but I think you should be able to use TPM2_Sign (which is already implemented) to produce an ECDAA signature using the commit value. You'll have to grab the commit counter from the Commit command and use it on the ECDAA scheme structure in the Sign command.

Bonus points: Call Sign again with the same counter value and expect it to fail :)

No need to validate the signature at this point, unless you really want to. I imagine that belongs as part of a larger body of ECDAA sample code that someone might work on (maybe you or someone else) as part of a separate effort.